synapse/handlers/federation_event.py

```diff
@@ -878,14 +861,14 @@ async def _resolve_state_at_missing_prevs(
                 "We can't get valid state history.",
                 affected=event_id,
             )
-        return state
+        return state_map
 
     async def _get_state_after_missing_prev_event(
```

Comment:
can we rename this to reflect that it returns event ids?

```suggestion
    async def _get_state_ids_after_missing_prev_event(
```

This is probably relevant to other places too.